modules/gapi/src/executor/gexecutor.cpp

```diff
@@ -382,16 +383,19 @@ void cv::gimpl::GExecutor::run(cv::gimpl::GRuntimeArgs &&args)
         magazine::resetInternalData(m_res, data);
     }
 
-    // Run the script
+    // Ask every IslandExecutable to prepate its internal states for new coming stream (4)
+    std::call_once(m_prep_flag, [this](){ this->prepareForNewStream(); });
```

Comment:
So this happens only once. Is this flag reset anywhere?

Reply:
No, this flag can't be reset as I've got it
By the way, this flag is different for any different object of `GExecutor`.
So, I might say that `prepareForNewStream()` is called once per each `GExecutor`.

Reply:
This is wrong.

It may be called at any moment for a GCompiled and NO internal recompilation may happen.

https://docs.opencv.org/4.x/d2/d2c/classcv_1_1GCompiled.html#ab3f7a54ed698a86ca6c63b4351eddb20

Reply:
Also. Should this happen in run or during the gexecutor construction? When it is constructed, valid `meta` should be there already/

Reply:
Yes, I mean, first `prepareForNewStream()` is called once per each `GExecutor`, which is not called by user.
Yes, it should be called upon the `GExecutor` construction, `std::call_once` is used only by that reason.
But, in the `GExecutor` execution model it seemed for me that it just fits better to be as 4th step:
    // Naive execution model is similar to current CPU (OpenCV) plugin
    // execution model:
    // 1. Allocate all internal resources first (NB - CPU plugin doesn't do it)
    // 2. Put input/output GComputation arguments to the storage
    // 3. For every Island, prepare vectors of input/output parameter descs
    **// 4. Ask every GIslandExecutable to prepate its internal states for a new stream**
    // 5. Iterate over a list of operations (sorted in the topological order)
    // 6. For every operation, form a list of input/output data objects
    // 7. Run GIslandExecutable
    // 8. writeBack

Reply:
Thanks, fixed!!

Reply:
Note I am not talking here about the initialization procedure.
`prepareForNewStream` **can** be called by user manually for the existing `GExecutor`.